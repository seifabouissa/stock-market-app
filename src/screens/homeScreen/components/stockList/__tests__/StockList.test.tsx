import ReactTestRenderer from 'react-test-renderer';
import StockList from '..';
import {Text} from 'react-native';

const stock = {
  ticker: 'AACG',
  name: 'ATA Creativity Global American Depositary Shares',
};

describe('Stock list component renders correctly', () => {
  it('Stock list component renders correctly', () => {
    const renderer = ReactTestRenderer.create(
      <StockList ticker={stock.ticker} name={stock.name} />,
    );
    const instance = renderer.root;
    const textElement = instance.findAllByType(Text);
    const ticker = textElement[0].props.children;
    const name = textElement[1].props.children;
    expect(ticker).toBe(stock.ticker);
    expect(name).toBe(stock.name);
  });

  it('Check if stock list component matches the snapshot', () => {
    const stockListRenderer = ReactTestRenderer.create(
      <StockList ticker={stock.ticker} name={stock.name} />,
    );
    const tree = stockListRenderer.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
