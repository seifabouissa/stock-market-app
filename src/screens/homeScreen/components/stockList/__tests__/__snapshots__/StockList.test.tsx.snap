// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stock list component renders correctly Check if stock list component matches the snapshot 1`] = `
<View
  style={
    {
      "backgroundColor": "#fff",
      "borderColor": "#0397C8",
      "borderRadius": 10,
      "borderWidth": 1,
      "flex": 1,
      "height": 100,
      "justifyContent": "center",
      "padding": 10,
    }
  }
>
  <Text
    style={
      {
        "color": "#0397C8",
        "fontSize": 20,
        "fontWeight": "bold",
        "marginBottom": 5,
        "textAlign": "center",
      }
    }
  >
    AACG
  </Text>
  <Text
    numberOfLines={1}
    style={
      {
        "color": "#000",
        "fontWeight": "bold",
        "textAlign": "center",
      }
    }
  >
    ATA Creativity Global American Depositary Shares
  </Text>
</View>
`;
